src/main/java/sx/blah/discord/api/internal/Requests.java

```diff
@@ -56,62 +58,60 @@
 	 */
 	public static final Requests GENERAL_REQUESTS = new Requests(null);
 
+	/**
+	 * The HTTP client requests are made on.
+	 */
+	private final OkHttpClient http;
+
 	/**
 	 * Used to send POST requests.
 	 */
-	public final Request POST;
+	public final DiscordRequest POST;
 	/**
 	 * Used to send GET requests.
 	 */
-	public final Request GET;
+	public final DiscordRequest GET;
 	/**
 	 * Used to send DELETE requests.
 	 */
-	public final Request DELETE;
+	public final DiscordRequest DELETE;
 	/**
 	 * Used to send PATCH requests.
 	 */
-	public final Request PATCH;
+	public final DiscordRequest PATCH;
 	/**
 	 * Used to send PUT requests.
 	 */
-	public final Request PUT;
-
-	/**
-	 * The client used for these requests.
-	 */
-	private final DiscordClientImpl client;
+	public final DiscordRequest PUT;
 
 	public Requests(DiscordClientImpl client) {
-		this.client = client;
-
-		POST = new Request(HttpPost.class, client);
-		GET = new Request(HttpGet.class, client);
-		DELETE = new Request(HttpDelete.class, client);
-		PATCH = new Request(HttpPatch.class, client);
-		PUT = new Request(HttpPut.class, client);
+		HttpLoggingInterceptor httpLogger = new HttpLoggingInterceptor(message -> Discord4J.LOGGER.debug(LogMarkers.API, message)).setLevel(HttpLoggingInterceptor.Level.HEADERS);
+		httpLogger.redactHeader("Set-Cookie");
+		httpLogger.redactHeader("Cookie");
+		httpLogger.redactHeader("Authorization");
+		http = new OkHttpClient.Builder().addInterceptor(httpLogger).build();
```

Comment:
We will want to test the scalability of this configuration. We may want to tweak `connectionPool()` and/or `dispatcher()`.